### PR TITLE
rust: begin converting workspace crates to use workspace dependencies

### PR DESCRIPTION
**Pull request by bmwill**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-genesis-builder/src/lib.rs [hunk 1]
```diff
@@ -628,7 +628,7 @@ impl Builder {
         fs::create_dir_all(&committee_dir)?;
 
         for (_pubkey, validator) in self.validators {
-            let validator_info_bytes = serde_yaml::to_vec(&validator)?;
+            let validator_info_bytes = serde_yaml::to_string(&validator)?;
```
Comment (after-ephemera) [inline crates/sui-genesis-builder/src/lib.rs:631]: Is this relevant to the workspace changes?

**Review by after-ephemera — APPROVED** [review]
I see that the rs changes were actually coming from the previous commit. Cool, let's try it out

File: Cargo.toml [hunk 2]
```diff
@@ -109,7 +109,6 @@ members = [
     "crates/sui-simulator",
     "crates/sui-snapshot",
     "crates/sui-source-validation",
-    "crates/sui-source-validation-service",
```
Comment (amnn) [inline Cargo.toml:112]: Is this intentional?  This is a new crate that @rvantonder is working on.

File: Cargo.toml [hunk 3]
```diff
@@ -191,11 +190,190 @@ opt-level = 1
...
+unescape = "0.1.0"
+url = "2.3.1"
+uuid = { version = "1.1.2", features = ["v4", "fast-rng"]}
+webpki = { version = "0.22.0", features = ["alloc", "std"] }
+x509-parser = "0.14.0"
+zstd = "0.12.3"
+versions = "4.1.0"
 
 # Move dependencies
+move-bytecode-verifier = { path = "external-crates/move/move-bytecode-verifier" }
```
Comment (amnn) [inline Cargo.toml:376]: These should not be workspace dependencies, they are part of the execution layer, and the `sui-execution` crate will multiplex between this and other versions of this crate -- I want to avoid other crates depending on these crates directly (there are some crates that are allowed to do that, but they should be the exception, and not the rule).

File: Cargo.toml [hunk 4]
```diff
@@ -204,7 +382,9 @@ move-core-types = { path = "external-crates/move/move-core/types", features = ["
 move-disassembler = { path = "external-crates/move/tools/move-disassembler" }
 move-package = { path = "external-crates/move/tools/move-package" }
 move-unit-test = { path = "external-crates/move/tools/move-unit-test" }
+move-stdlib = { path = "external-crates/move/move-stdlib" }
```
Comment (amnn) [inline Cargo.toml:385]: ditto, execution-layer.

File: Cargo.toml [hunk 5]
```diff
@@ -204,7 +382,9 @@ move-core-types = { path = "external-crates/move/move-core/types", features = ["
...
 move-vm-config = { path = "external-crates/move/move-vm/config" }
+move-vm-runtime = { path = "external-crates/move/move-vm/runtime" }
```
Comment (amnn) [inline Cargo.toml:387]: ditto, execution-layer.

File: Cargo.toml [hunk 6]
```diff
@@ -225,12 +405,92 @@ anemo-build = { git = "https://github.com/mystenlabs/anemo.git", rev = "0f0ae8d8
...
+narwhal-network = { path = "narwhal/network" }
+narwhal-node = { path = "narwhal/node" }
+narwhal-primary = { path = "narwhal/primary" }
+narwhal-storage = { path = "narwhal/storage" }
+narwhal-test-utils = { path = "narwhal/test-utils" }
+narwhal-types = { path = "narwhal/types" }
+narwhal-worker = { path = "narwhal/worker" }
+
+sui-execution = { path = "sui-execution" }
+sui-move-natives = { path = "sui-execution/latest/sui-move-natives", package = "sui-move-natives-latest" }
```
Comment (amnn) [inline Cargo.toml:494]: ditto, execution-layer for everything below this line (inclusive).  But `sui-execution` can stay here.

File: crates/sui-sdk/Cargo.toml [hunk 7]
```diff
@@ -7,47 +7,46 @@ publish = false
...
-jsonrpsee = { git="https://github.com/wlmyng/jsonrpsee.git", rev= "b1b300784795f6a64d0fcdf8f03081a9bc38bde8", features = ["http-client"] }
-serde = { version = "1.0.144", features = ["derive"] }
-serde_with = { version = "2.1.0", features = ["hex"] }
-serde_json = "1.0.88"
-futures-core = "0.3.21"
-futures = "0.3.23"
-tokio = { workspace = true }
-bcs = "0.1.4"
-thiserror = "1.0.37"
+anyhow.workspace = true
```
Comment (amnn) [inline crates/sui-sdk/Cargo.toml:10]: I thought we generally wanted to extricate `sui-sdk` from the workspace and `workspace-hack` (or does not using `workspace-hack` not preclude using the `workspace` for dependencies?)

Comment (amnn) [reply crates/sui-sdk/Cargo.toml:10]: Also cc @stefan-mysten for this discussion as you'll be looking into distributing `sui-sdk` on crates.io etc.

File: sui-execution/latest/sui-adapter/src/programmable_transactions/execution.rs [hunk 8]
```diff
@@ -831,6 +831,7 @@ fn vm_move_call(
     Ok(result)
 }
 
+#[allow(clippy::extra_unused_type_parameters)]
 fn deserialize_modules<Mode: ExecutionMode>(
```
Comment (amnn) [inline sui-execution/latest/sui-adapter/src/programmable_transactions/execution.rs:835]: huh... I think we could get rid of this type parameter, but I'm happy to do that as a follow-up.

**Review by amnn — CHANGES_REQUESTED** [review]
Requesting changes mostly for the `sui-execution` related crates -- (I may be missing context and everything is fine, but I'm just being extra careful until I can write a test in CI that makes sure cutting new execution versions doesn't break).

File: Cargo.toml (hunk 2 above)
Comment (bmwill) [reply Cargo.toml:112]: Interesting i didn't intend to remove that...I wonder what happened. One thing that i did notice is that that crate uses `actix` @rvantonder can you please remove the use of the actix crate? We already use `axum` for web services and should not be bringing in yet another web framework for one service.

File: Cargo.toml (hunk 3 above)
Comment (bmwill) [reply Cargo.toml:376]: Making them "workspace dependencies" would not impact this at all, afaik

File: Cargo.toml (hunk 4 above)
Comment (bmwill) [reply Cargo.toml:385]: See my other comment,, listing these as "workspace dependencies" doesn't implicitly make things depend on these crates, it just makes the usage of them a bit easier elsewhere in the codebase. Listing these here does not impact any of the work you want to do with the execution layer, id argue it makes it easier

File: crates/sui-sdk/Cargo.toml (hunk 7 above)
Comment (bmwill) [reply crates/sui-sdk/Cargo.toml:10]: The whole post of this PR is to make it very easy to continue having the sdk be a part of the workspace while eliminating the larger down sides of workspace-hack

Comment (amnn) [reply crates/sui-sdk/Cargo.toml:10]: Awesome, thanks for clarifying!

Comment (bmwill) [reply crates/sui-sdk/Cargo.toml:10]: I'm happy if we want to do something different for the sdk as well, this was just me trying to be helpful towards that effort :)
